scripts/scr_initialize_custom/scr_initialize_custom.gml

```diff
@@ -316,9 +321,12 @@ function scr_initialize_custom() {
 	    ship_names+=string(ship[v])+"|";
 	}
 
-	var j,f;j=0;f=0;
-	repeat(battle_barges+strike_cruisers+gladius+hunters){f+=1;j=0;
-	    repeat(30){j+=1;if (ship_uid[f]=ship_uid[j]) and (f!=j) then ship_uid[j]=floor(random(99999999))+1;}
+	var j=0,f=0;
+	var total_ship_count = battle_barges+strike_cruisers+gladius+hunters;
+	for (f=1;f<total_ship_count+1;f++){
+	    for(j=1;j<31;j++){
+	    	if (ship_uid[f]=ship_uid[j]) and (f!=j) then ship_uid[j]=floor(random(99999999))+1;
```

Comment:
does this generate a random id for the ship?

Reply:
really not sure what this does all i did was swap out the repeat, shall investigate

Reply:
attempts to generate an id although looking through he code i don't think it's ever actually utilised